clang/test/BoundsSafety/CodeGen/array_subscript_agg.c

```diff
@@ -0,0 +1,1620 @@
+// LEGACY-NEXT:    [[TMP0:%.*]] = load ptr, ptr [[PTR_ADDR]], align 8
+// LEGACY-NEXT:    [[TMP1:%.*]] = load i32, ptr [[COUNT_ADDR]], align 4
+// LEGACY-NEXT:    [[IDX_EXT:%.*]] = sext i32 [[TMP1]] to i64
+// LEGACY-NEXT:    [[ADD_PTR:%.*]] = getelementptr inbounds [[STRUCT_FOO:%.*]], ptr [[TMP0]], i64 [[IDX_EXT]]
+// LEGACY-NEXT:    [[TMP2:%.*]] = getelementptr inbounds nuw %"__bounds_safety::wide_ptr.bidi_indexable", ptr [[AGG_TEMP]], i32 0, i32 0
+// LEGACY-NEXT:    store ptr [[TMP0]], ptr [[TMP2]], align 8
+// LEGACY-NEXT:    [[TMP3:%.*]] = getelementptr inbounds nuw %"__bounds_safety::wide_ptr.bidi_indexable", ptr [[AGG_TEMP]], i32 0, i32 1
+// LEGACY-NEXT:    store ptr [[ADD_PTR]], ptr [[TMP3]], align 8
+// LEGACY-NEXT:    [[TMP4:%.*]] = getelementptr inbounds nuw %"__bounds_safety::wide_ptr.bidi_indexable", ptr [[AGG_TEMP]], i32 0, i32 2
+// LEGACY-NEXT:    store ptr [[TMP0]], ptr [[TMP4]], align 8
+// LEGACY-NEXT:    [[WIDE_PTR_PTR_ADDR:%.*]] = getelementptr inbounds nuw %"__bounds_safety::wide_ptr.bidi_indexable", ptr [[AGG_TEMP]], i32 0, i32 0
+// LEGACY-NEXT:    [[WIDE_PTR_PTR:%.*]] = load ptr, ptr [[WIDE_PTR_PTR_ADDR]], align 8
+// LEGACY-NEXT:    [[TMP5:%.*]] = load i32, ptr [[IDX_ADDR]], align 4
+// LEGACY-NEXT:    [[IDXPROM:%.*]] = sext i32 [[TMP5]] to i64
+// LEGACY-NEXT:    [[ARRAYIDX:%.*]] = getelementptr [[STRUCT_FOO]], ptr [[WIDE_PTR_PTR]], i64 [[IDXPROM]]
+// LEGACY-NEXT:    [[WIDE_PTR_UB_ADDR:%.*]] = getelementptr inbounds nuw %"__bounds_safety::wide_ptr.bidi_indexable", ptr [[AGG_TEMP]], i32 0, i32 1
+// LEGACY-NEXT:    [[WIDE_PTR_UB:%.*]] = load ptr, ptr [[WIDE_PTR_UB_ADDR]], align 8
+// LEGACY-NEXT:    [[WIDE_PTR_LB_ADDR:%.*]] = getelementptr inbounds nuw %"__bounds_safety::wide_ptr.bidi_indexable", ptr [[AGG_TEMP]], i32 0, i32 2
+// LEGACY-NEXT:    [[WIDE_PTR_LB:%.*]] = load ptr, ptr [[WIDE_PTR_LB_ADDR]], align 8
+// LEGACY-NEXT:    [[TMP6:%.*]] = icmp ult ptr [[ARRAYIDX]], [[WIDE_PTR_UB]], !annotation [[META3]]
+// LEGACY-NEXT:    br i1 [[TMP6]], label %[[CONT:.*]], label %[[TRAP:.*]], !annotation [[META3]]
+// LEGACY:       [[TRAP]]:
+// LEGACY-NEXT:    call void @llvm.ubsantrap(i8 25) #[[ATTR3]], !annotation [[META3]]
+// LEGACY-NEXT:    unreachable, !annotation [[META3]]
+// LEGACY:       [[CONT]]:
+// LEGACY-NEXT:    [[TMP7:%.*]] = icmp uge ptr [[ARRAYIDX]], [[WIDE_PTR_LB]], !annotation [[META4]]
+// LEGACY-NEXT:    br i1 [[TMP7]], label %[[CONT2:.*]], label %[[TRAP1:.*]], !annotation [[META4]]
+// LEGACY:       [[TRAP1]]:
+// LEGACY-NEXT:    call void @llvm.ubsantrap(i8 25) #[[ATTR3]], !annotation [[META4]]
+// LEGACY-NEXT:    unreachable, !annotation [[META4]]
+// LEGACY:       [[CONT2]]:
+// LEGACY-NEXT:    [[X:%.*]] = getelementptr inbounds nuw [[STRUCT_FOO]], ptr [[ARRAYIDX]], i32 0, i32 0
+// LEGACY-NEXT:    [[TMP8:%.*]] = load i32, ptr [[X]], align 4
+// LEGACY-NEXT:    ret i32 [[TMP8]]
+//
+int access_Foo_member_cb(struct Foo* __counted_by(count) ptr, int idx, int count) {
+  return ptr[idx].x;
```

Comment:
Could we add a test when the member `x` is also an aggregate?

Reply:
Will do

Reply:
Done see the `access_MemberIsAgg_member_bidi` test case.